src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -368,24 +369,30 @@ public void sendActionListenerOnFailureRequest(TransportService transportService
     }
 
     /**
-     * Requests the environment setting values from OpenSearch for the corresponding component settings. The result will be handled by a {@link EnvironmentSettingsResponseHandler}.
+     * Requests the environment settings from OpenSearch. The result will be handled by a {@link EnvironmentSettingsResponseHandler}.
      *
-     * @param componentSettings The component setting that correspond to the values provided by the environment settings
      * @param transportService  The TransportService defining the connection to OpenSearch.
+     * @return A Setting object from the OpenSearch Node environment
      */
-    public void sendEnvironmentSettingsRequest(TransportService transportService, List<Setting<?>> componentSettings) {
+    public Settings sendEnvironmentSettingsRequest(TransportService transportService) {
```

Comment:
When is this called? If we're starting up the extension before OpenSearch, there are no settings to fetch.  Should this be triggered as part of initialization?